pkg/webhook/controlplane/ensurer.go

```diff
@@ -149,6 +148,133 @@ func keepNodeNetworkEnvVarIfPresentInOldDeployment(new, old *appsv1.Deployment,
 	}
 }
 
+// EnsureAdditionalProvisionUnits ensures that additional required system units are added, that are required during provisioning.
+func (e *ensurer) EnsureAdditionalProvisionUnits(ctx context.Context, gctx gcontext.GardenContext, new, _ *[]extensionsv1alpha1.Unit) error {
+
+	cluster, err := gctx.GetCluster(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, worker := range cluster.Shoot.Spec.Provider.Workers {
+		if worker.DataVolumes != nil {
+			switch worker.Machine.Image.Name {
+			case "flatcar":
+				extensionswebhook.AppendUniqueUnit(new, extensionsv1alpha1.Unit{
+					Name:    "lvm-setup.service",
+					Enable:  ptr.To(true),
+					Command: ptr.To(extensionsv1alpha1.CommandStart),
+					Content: ptr.To(`[Unit]
+        Description=LVM Setup
+        ConditionFirstBoot=yes
+        DefaultDependencies=no
+        Before=local-fs-pre.target
+        [Service]
+        Type=oneshot
+        Restart=on-failure
+        RemainAfterExit=yes
+        ExecStart=/opt/bin/lvm.sh
+        [Install]
+        WantedBy=multi-user.target
+`),
+				})
+
+				extensionswebhook.AppendUniqueUnit(new, extensionsv1alpha1.Unit{
+					Name:    "var-lib-docker.mount",
+					Enable:  ptr.To(true),
+					Command: ptr.To(extensionsv1alpha1.CommandStart),
+					Content: ptr.To(`[Unit]
+        Description=Mount LVM to docker dir
+        After=lvm-setup.service
+        Before=docker.service
+        [Mount]
+        What=/dev/vg-docker/vol_docker
+        Where=/var/lib/docker
+        Type=ext4
+        Options=defaults
+        [Install]
+        WantedBy=local-fs.target
+`),
+				})
+
+				return nil
+			}
+
+		}
+	}
+	return nil
+}
+
+// EnsureAdditionalProvisionFiles ensures that additional required system files are added, that are required during provisioning.
```

Comment:
we should include a short decription of what this script is doing.

something like:
```
Creates a lvm from all remaining non-boot disks.
```
And also on the function on top that the created volume will be mounted to the containerd/docker lib dir